src/js/components/TextArea/TextArea.js

```diff
@@ -1,14 +1,42 @@
 import React, { Component } from 'react';
 import { compose } from 'recompose';
 
+import { Keyboard } from '../Keyboard';
 import { withFocus, withForwardRef, withTheme } from '../hocs';
 
 import { StyledTextArea } from './StyledTextArea';
 
 class TextArea extends Component {
+  static getDerivedStateFromProps(nextProps, prevState) {
+    const { forwardRef } = nextProps;
+    const { inputRef } = prevState;
+    const nextInputRef = forwardRef || inputRef;
+    if (nextInputRef !== inputRef) {
+      return { inputRef: nextInputRef };
+    }
+    return null;
+  }
+
+  state = {
+    inputRef: React.createRef(),
+  };
+
+  onEsc = event => {
+    const { inputRef } = this.state;
+    if (event.keyCode === 27 && document.activeElement === inputRef.current) {
```

Comment:
why we have to check for the keyCode? I think onEsc can always do `event.nativeEvent.stopImmediatePropagation()`. I don't think we need to blur the input.